### PR TITLE
Allow repeated calls to connect and disconnect

### PR DESCRIPTION
**Pull request by sondrelg**
Replaces assertions with early returns as discussed in https://github.com/encode/databases/issues/359.

### CODE REVIEW

**Comment by sondrelg** [comment]
Looks like the test suite fails from a drop in coverage. I'll happily add tests for this, but do you think this is otherwise mergable seeing that no tests actually failed here @aminalaee? 🙂

**Comment by aminalaee** [comment]
@sondrelg Thanks you for this. Yes, it looks good to me , and I don't think it would break anything.
I'll keep it open for a day or two to see if anyone has any objections though.

**Comment by aminalaee** [comment]
Closes #359

**Comment by sondrelg** [comment]
Ok cool, I'll add a test or two!

File: tests/test_databases.py [hunk 1]
```diff
@@ -737,6 +737,13 @@ async def test_connect_and_disconnect(database_url):
     await database.disconnect()
     assert not database.is_connected
 
+    # Test duplicate connect and disconnect
+    await database.connect()
+    await database.connect()
+    assert database.is_connected
+    await database.disconnect()
+    await database.disconnect()
+    assert not database.is_connected
```
Comment (aminalaee) [inline tests/test_databases.py:747]:
```suggestion


```

**Review by aminalaee — CHANGES_REQUESTED** [review]

**Review by aminalaee — APPROVED** [review]
@sondrelg  Thanks for this.

**Pull request merged**